src/realm/util/bind_ptr.hpp

```diff
@@ -260,6 +260,12 @@ class bind_ptr : public bind_ptr_base {
     friend class bind_ptr;
 };
 
+// Deduction guides
```

Comment:
I don't object to having this in here, but how is this related to the rest of the PR?

Reply:
It's not. I had been checking if clang 14 had any interesting new warnings (it doesn't seem to) and accidentally included this commit when rebasing the branch.